cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ApiCredentials.java

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model.cruisecontrol;
+
+import io.fabric8.kubernetes.api.model.OwnerReference;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.api.kafka.model.kafka.cruisecontrol.CruiseControlResources;
+import io.strimzi.api.kafka.model.kafka.cruisecontrol.CruiseControlSpec;
+import io.strimzi.api.kafka.model.kafka.cruisecontrol.HashLoginServiceApiUsers;
+import io.strimzi.operator.cluster.model.ModelUtils;
+import io.strimzi.operator.common.InvalidConfigurationException;
+import io.strimzi.operator.common.Util;
+import io.strimzi.operator.common.model.InvalidResourceException;
+import io.strimzi.operator.common.model.Labels;
+import io.strimzi.operator.common.model.PasswordGenerator;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.AUTH_FILE_KEY;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.HEALTHCHECK_PASSWORD_KEY;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.HEALTHCHECK_USERNAME;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.REBALANCE_OPERATOR_PASSWORD_KEY;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.REBALANCE_OPERATOR_USERNAME;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.TOPIC_OPERATOR_PASSWORD_KEY;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.TOPIC_OPERATOR_USERNAME;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.TOPIC_OPERATOR_USERNAME_KEY;
+
+/**
+ * Uses information in a Kafka Custom Resource to generate a API credentials configuration file to be used for
+ * authenticating to Cruise Control's REST API.
+ */
+public class ApiCredentials {
+    // Regex to match an entry in Jetty's HashLoginService's file format: username: password, rolename
+    private static final Pattern HASH_LOGIN_SERVICE_PATTERN = Pattern.compile("^[\\w-]+\\s*:\\s*\\w+\\s*,\\s*\\w+\\s*$");
```

Comment:
This is kind of related to my question on the `HashLoginServiceApiUsers`, we are not really open to future types unless refactoring the existing code.
I mean this `ApiCredentials` class should be generic to support even future service api users types.
Putting `HASH_LOGIN_SERVICE_PATTERN` here seems to tight to the only current implementation we have so maybe it should be moved to the `HashLoginServiceApiUsers` class.
Of course, it means that where you are using the constant (within `parseEntriesFromString` method) you should get which type are we using and parsing the right way. Maybe better the `parseEntriesFromString` itself should be part of the api abstract base model class (see my suggestion around having a `CruiseControlServiceApiUsers` or `ServiceApiUsers`) so that you can have a different implemented parsing logic based on the actual implementation of the service api users type.

Reply:
Refactored the code, let me know what you think!